src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientInternal.java

```diff
@@ -241,14 +241,13 @@ public SnowflakeStreamingIngestClientInternal(
 
     this.snowflakeServiceClient = new SnowflakeServiceClient(this.httpClient, this.requestBuilder);
 
-    this.storageManager =
-        parameterProvider.isEnableIcebergStreaming()
-            ? new SubscopedTokenExternalVolumeManager(
-                this.role, this.name, this.snowflakeServiceClient)
-            : new InternalStageManager(
-                isTestMode, this.role, this.name, this.snowflakeServiceClient);
-
     try {
+      this.storageManager =
```

Comment:
this got moved into the try catch?

Reply:
yes, otherwise the background threads will keep running, it will be cleaned up by the `cleanUpResources` call in the catch clause